Add ADR_006: Geographic Sharding (#10)
selfdriventeam/kata: doc/arc/adr_006.md

> # Sharding & Routing per Geographic Region

Reviewer: Let's add the link back to the root readme. See the other ADRs fort examples

Author: ```suggestion
# [ADR_006](../../README.md) Sharding & Routing per Geographic Region
```

> Date: Nov 1 2020

Reviewer: Silly, but let's be consistent w/ date format

Author: ```suggestion
Date: 11/1
```

## Status

Accepted

## Context

When trying to build a robust and highly available system, it is important to consider how to manage users from different regions. Sharding users to different deployments of a service based on their geographic location gives some natural load balancing, and ensures that users are not heavily impacted by latency to distant servers.

## Decision

> We have chosen to have a singular deployment of the pharmacy food system.
> 
> Pharmacyfood's goals at this time are to operate and expland within a single city in the near term. This does not justify the complication of managing multiple different deployments in one or more datacenters,  or dealing with sharding users across infrastructure.

Author: ```suggestion
We have chosen to have a singular deployment of the farmacy food system.

Farmacyfood's goals at this time are to operate and expand within a single city in the near term. This does not justify the complication of managing multiple different deployments in one or more datacenters,  or dealing with sharding users across infrastructure.
```

## Consequences

> Should Pharmacyfood expand to multiple citys across the united states, the nature of viewing/ordering meals and recieving recomendations will likely not be heavily impacted by higher latency operations due to geographic distance.

Author: ```suggestion
Should farmacyfood expand to multiple cities across the united states, the nature of viewing/ordering meals and receiving recommendations will likely not be heavily impacted by higher latency operations due to geographic distance.
```

> Business wise, it would be a very great problem for Pharmacyfoods to have if a singular deployment is not scaling well enough. That bridge can be crossed later.

Author: ```suggestion
Business wise, it would be a very great problem for farmacyfoods to have if a singular deployment is not scaling well enough. That bridge can be crossed later.
```
